src/components/VSelect/VSelect.js

```diff
@@ -400,18 +400,30 @@ export default {
     },
     genList () {
       // If there's no slots, we can use a cached VNode to improve performance
-      if (this.$slots['no-data']) {
+      if (this.$slots['no-data'] || this.$slots['before-list'] || this.$slots['after-list']) {
         return this.genListWithSlot()
       } else {
         return this.staticList
       }
     },
     genListWithSlot () {
-      return this.$createElement(VSelectList, this.listData, [
-        this.$createElement('template', {
+      const slots = []
+      if (this.$slots['no-data']) {
```

Comment:
how about `const slots = ['before-list', 'no-data', 'after-list'].map(slotName => this.$createElement('template', ...)`?